src/libraries/Common/src/System/IO/PathInternal.CaseSensitivity.cs

```diff
@@ -31,13 +31,30 @@ internal static StringComparison StringComparison
         /// <summary>
         /// Determines whether the file system is case sensitive.
         /// </summary>
+        private static bool GetIsCaseSensitive()
+        {
+#if NET6_0_OR_GREATER
+            // Return a constant for mobile platforms without resorting to I/O
+            if (OperatingSystem.IsMacOS() || OperatingSystem.IsMacCatalyst() || OperatingSystem.IsIOS() || OperatingSystem.IsTvOS() || OperatingSystem.IsWatchOS())
+                return true;
+            if (OperatingSystem.IsBrowser())
+                return false;
+            if (OperatingSystem.IsAndroid())
+                return false;
+#endif
+            return GetIsCaseSensitiveByProbing();
+        }
+
+        /// <summary>
+        /// Determines whether the file system is case sensitive by creating a file in a temp folder and observing the result.
+        /// </summary>
         /// <remarks>
         /// Ideally we'd use something like pathconf with _PC_CASE_SENSITIVE, but that is non-portable,
         /// not supported on Windows or Linux, etc. For now, this function creates a tmp file with capital letters
         /// and then tests for its existence with lower-case letters.  This could return invalid results in corner
         /// cases where, for example, different file systems are mounted with differing sensitivities.
         /// </remarks>
-        private static bool GetIsCaseSensitive()
+        private static bool GetIsCaseSensitiveByProbing()
```

Comment:
I think that it would be great to use the old implementation as a test of the new one.

You could introduce a new test in [System.IO.FileSystem.Tests.csproj](https://github.com/dotnet/runtime/blob/main/src/libraries/System.IO.FileSystem/tests/System.IO.FileSystem.Tests.csproj) include the `.cs` file like this:


```xml
    <Compile Include="$(CommonPath)System\IO\PathInternal.CaseSensitivity.cs"
             Link="Common\System\IO\PathInternal.CaseSensitivity.cs" />
```

 and move `GetIsCaseSensitiveByProbing` there to get the "expected".

Something like this:

```cs
[Fact]
public void GetIsCaseSensitiveReturnsSameValueAsProbing()
{
    Assert.Equal(GetIsCaseSensitiveByProbing(), PathInternal.IsCaseSensitive);
}
```

Reply:
That would be unreliable test. The temp directory on Unix may live on a case insensensitive volume.

Reply:
> That would be unreliable test. The temp directory on Unix may live on a case insensensitive volume.

So perhaps it should be using current directory instead of temp? I know that it would still be far from perfect but I would prefer to have a single non-perfect test rather than no tests at all.

Reply:
Current directory has the same problem. We can have a test like that, but it should be marked as outerloop.

Reply:
> We can have a test like that, but it should be marked as outerloop.

Sounds good to me.

Reply:
> So perhaps it should be using current directory instead of temp

That might fail on mobile for trivial reasons - I'm not sure the tests run from a writable directory. I'll give it a try and see.